### PR TITLE
Use `dyn trait` everywhere

### PR DESCRIPTION
**Pull request by Manishearth**
Based on https://github.com/rust-lang/rust/pull/48461

do not land unless both are reviewed.

### CODE REVIEW

File: src/librustc_mir/transform/mod.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ pub macro run_passes($tcx:ident, $mir:ident, $def_id:ident, $suite_index:expr; $
             promoted
         };
         let mut index = 0;
-        let mut run_pass = |pass: &dyn MirPassPassPass| {
+        let mut run_pass = |pass: &dyn MirPass| {
```
Comment (nikomatsakis) [inline src/librustc_mir/transform/mod.rs]: what's up with these fixups?

File: src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs [hunk 2]
```diff
@@ -176,7 +176,7 @@ impl<'gcx, 'tcx> UseFinder<'gcx, 'tcx> {
         None
     }
 
-    fn def_use(&self, location: Location, thing: &MirVisitable<'tcx>) -> (bool, bool) {
+    fn def_use(&self, location: Location, thing: &dyn MirVisitable<'tcx>) -> (bool, bool) {
```
Comment (nikomatsakis) [inline src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs]: this was all automated, right?

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
@Manishearth r=me, but i'd like to know to what extent this was done automatically with rustfix (the commit names suggest it was), and what drove the need for the fixups -- do those represent bugs in the compiler suggestions?

File: src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs (hunk 2 above)
Comment (Manishearth) [reply src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs]: Yeah

File: src/librustc_mir/transform/mod.rs (hunk 1 above)
Comment (Manishearth) [reply src/librustc_mir/transform/mod.rs]: Macros are involved and rustfix broke. There's a bug filed for this.

**Comment by Manishearth** [comment]
It's because of macros; we should be using the non-machine-applicable flag when macros are involved, and/or make rustfix better at this.

In these cases rustfix would be able to help if it understood when multiple suggestions are the same textual suggestion (there's a bug filed)

**Comment by Manishearth** [comment]
@bors r=nmatsakis

The commits with "rustfix" in their message were automated, fwiw

**Comment by Manishearth** [comment]
@bors p=2

potential to bitrot since it's a major lint

**Pull request merged**